pkg/virt-controller/watch/vm.go

```diff
@@ -297,6 +297,7 @@ func (c *VMController) execute(key string) error {
 		return nil
 	}
 	vm := obj.(*virtv1.VirtualMachine)
+	vm = vm.DeepCopy()
```

Comment:
Just curious: where was the actual cache corruption? I tried to track it and the code below seems to do deep copy where needed. But maybe I missed that...

Reply:
indeed. nice catch :+1:

Reply:
@vasiliy-ul 
[Here](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L2974) as well.
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L875
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L2944
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L2957

Reply:
> @vasiliy-ul [Here](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L2974) as well. https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L875 

:+1: 

> https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L2944 https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L2957

`c.removeVMFinalizer` and `c.addVMFinalizer` do a deep copy internally.

Reply:
Ah right, should have looked at my notes better... They do it but they do not need to. I will post a patch for it.